### ISSUE TITLE
No such image error on successful docker pull

### ISSUE THREAD

**Issue by fdawg4l** [vmware/vic#1778]
Reported by @mreferre.  We suspect it's the datastore name, datastore path, or both being passed to `vic-machine create`

```
docker --tls run -it busybox
Unable to find image 'busybox:latest' locally
latest: Pulling from library/busybox
a3ed95caeb02: Pull complete 
8ddc19f16526: Pull complete 
Digest: sha256:65ce39ce3eb0997074a460adfb568d0b9f0f6a4392d97b6035630c9d7bf92402
Status: Downloaded newer image for library/busybox:latest
docker: Error: No such image: busybox.
```

`/vic-machine create --target https://'administrator@vsphere.local:Vmware123!'@192.168.8.211 -i  sharedstorage --name VCH1 --external-network External --bridge-network BridgeVCH1”`

**Comment by fdawg4l** [comment]
cc @gigawhitlocks 

Ian, I think this issue goes away with your change (#1678).  I'm going to assign this to you to try it with your change and close it if the issue goes away.  If not, feel free to debug or bounce it back to me to take a look at.

**Comment by mlh78750** [comment]
I'm seeing same for create command on VC 6.5 beta 3:

`Bauxite:~/vicbin/vic-3538 $ docker -H 10.160.19.62:2376 --tls create busybox
Unable to find image 'busybox:latest' locally
latest: Pulling from library/busybox
a3ed95caeb02: Pull complete
8ddc19f16526: Pull complete
Digest: sha256:65ce39ce3eb0997074a460adfb568d0b9f0f6a4392d97b6035630c9d7bf92402
Status: Downloaded newer image for library/busybox:latest
Error: No such image: busybox`

Install command was: 
`Bauxite:~/vicbin/vic-3538 $ ./vic-machine-darwin create --target 10.160.64.44  --image-datastore datastore1 --name vic-demo --user Administrator@vsphere.local --password 'Admin!23'  --compute-resource /ha-datacenter/host/10.160.6.233/Resources --bridge-network bridge --external-network vm-network`

**Comment by mlh78750** [comment]
Also, after you do this on VC 6.5 beta 3, you cannot delete the files in the datastore over the API. I have to login to the esx and rm -rf. Something is getting messed up on the host side. I can provide the logins/ip's for my setup on nimbus if someone wants to look at this.

**Comment by mlh78750** [comment]
After the failure a vic-machine delete has the following error:
`Bauxite:~/vicbin/vic-3538 $ ./vic-machine-darwin delete  --target 10.160.64.44  --name vic-demo --user Administrator@vsphere.local --password 'Admin!23'  --compute-resource /ha-datacenter/host/10.160.6.233/Resources --force
INFO[2016-08-02T15:10:41-05:00] ### Removing VCH ####
INFO[2016-08-02T15:10:48-05:00] Removing VMs
INFO[2016-08-02T15:10:49-05:00] Removing images
ERRO[2016-08-02T15:10:50-05:00] Operation failed: Cannot delete file [datastore1] VIC/42166ba8-5651-b807-0931-b73f32481a04/images - (&{{} %!!(MISSING)s(*types.CannotDeleteFile=&{{{{<nil> [{{} backtrace [{{} opId 1c458865}] [context]zKq7AZECAAAAAOtlPwAcdnB4ZAAAkWMybGlidm1hY29yZS5zbwAAYgwdANFDGYFCawgBbGlidmltLXR5cGVzLnNvAIFpMToBgYO+OgGBh846AQJ90BJsaWJ2bW9taS5zbwACmdISA84Da3ZweGQAA90oawPBp2oDNfxqAj9vFAIAZBQE8YAQbGlidnB4YXBpLXR5cGVzLnNvAAN/UvSBAJcSAQPskoMDgkmCA6Y8ggP5ZYIDhsWCAHmuKAC7sSgATdgyBUR0AGxpYnB0aHJlYWQuc28uMAAGLY4ObGliYy5zby42AA==[/context]}]}} [datastore1] VIC/42166ba8-5651-b807-0931-b73f32481a04/images}}) Cannot delete file [datastore1] VIC/42166ba8-5651-b807-0931-b73f32481a04/images})
ERRO[2016-08-02T15:10:52-05:00] Operation failed: Cannot delete file [datastore1] VIC/42166ba8-5651-b807-0931-b73f32481a04 - (&{{} %!!(MISSING)s(*types.CannotDeleteFile=&{{{{<nil> [{{} backtrace [{{} opId 6c9f53ff}] [context]zKq7AZECAAAAAOtlPwAcdnB4ZAAAkWMybGlidm1hY29yZS5zbwAAYgwdANFDGYFCawgBbGlidmltLXR5cGVzLnNvAIFpMToBgYO+OgGBh846AQJ90BJsaWJ2bW9taS5zbwACmdISA84Da3ZweGQAA90oawPBp2oDNfxqAj9vFAIAZBQE8YAQbGlidnB4YXBpLXR5cGVzLnNvAAN/UvSBAJcSAQPskoMDgkmCA6Y8ggP5ZYIDhsWCAHmuKAC7sSgATdgyBUR0AGxpYnB0aHJlYWQuc28uMAAGLY4ObGliYy5zby42AA==[/context]}]}} [datastore1] VIC/42166ba8-5651-b807-0931-b73f32481a04}}) Cannot delete file [datastore1] VIC/42166ba8-5651-b807-0931-b73f32481a04})
INFO[2016-08-02T15:10:52-05:00] Removing volume stores...
INFO[2016-08-02T15:10:52-05:00] Removing appliance VM network devices
INFO[2016-08-02T15:10:56-05:00] Removing VCH vSphere extension
INFO[2016-08-02T15:11:04-05:00] Removing Resource Pool "vic-demo"
INFO[2016-08-02T15:11:05-05:00] Completed successfully`

**Comment by fdawg4l** [comment]
From @mreferre 

> it turned out I was using the vic-machine version included with the 0.4.0 release while I was using ISOs compiled from the master branch. Now I have (re)BUILT ALL (ISOs + vic-machine) from source and I can start busybox just fine (albeit slowly — but I am digressing). Sorry for the useless firedrill

**Comment by mlh78750** [comment]
I installed and deleted with the binaries versions and didn't mix. There is still a problem here with VC targets.

**Comment by corrieb** [comment]
We should look at why our integration tests failed to pick this up

**Comment by gigawhitlocks** [comment]
I believe this is resolved by #1678

**Comment by stuclem** [comment]
This is fixed. Removing kind/note.
